From c4421fce41d592f839654aedecb6d52ca753c493 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:05 -0400
Subject: [infiniband] core: Make ib_addr a core IB module

Message-id: <f06fc76260c9fca5a209d1b1aed7bbf7b7592a50.1410425016.git.avadai@redhat.com>
Patchwork-id: 93577
O-Subject: [Patch RHEL7.1 V1 009/197] IB/core: Make ib_addr a core IB module
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Matan Barak <matanb@mellanox.com>

IP based addressing introduces the usage of rdma_addr_find_dmac_by_grh()
within ib_core.  Since this function is declared in ib_addr, ib_addr
should be a part of the core INFINIBAND modules, rather than
INFINIBAND_ADDR_TRANS.

Signed-off-by: Matan Barak <matanb@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 2f85d24e604c1532723c4b5791816b533baed2c3)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/Kconfig b/drivers/infiniband/Kconfig
index 5ceda71..3c101a0 100644
--- a/drivers/infiniband/Kconfig
+++ b/drivers/infiniband/Kconfig
@@ -3,6 +3,8 @@ menuconfig INFINIBAND
  depends on PCI || BROKEN
  depends on HAS_IOMEM
  depends on NET
+ depends on INET
+ depends on m || IPV6 != m
  ---help---
    Core support for InfiniBand (IB).  Make sure to also select
    any protocols you wish to use as well as drivers for your
@@ -38,8 +40,7 @@ config INFINIBAND_USER_MEM
 
 config INFINIBAND_ADDR_TRANS
  bool
- depends on INET
- depends on !(INFINIBAND = y && IPV6 = m)
+ depends on INFINIBAND
  default y
 
 source "drivers/infiniband/hw/mthca/Kconfig"
diff --git a/drivers/infiniband/core/Makefile b/drivers/infiniband/core/Makefile
index c8bbaef..3ab3865 100644
--- a/drivers/infiniband/core/Makefile
+++ b/drivers/infiniband/core/Makefile
@@ -1,8 +1,9 @@
-infiniband-$(CONFIG_INFINIBAND_ADDR_TRANS) := ib_addr.o rdma_cm.o
+infiniband-$(CONFIG_INFINIBAND_ADDR_TRANS) := rdma_cm.o
 user_access-$(CONFIG_INFINIBAND_ADDR_TRANS) := rdma_ucm.o
 
 obj-$(CONFIG_INFINIBAND) +=  ib_core.o ib_mad.o ib_sa.o \
-     ib_cm.o iw_cm.o $(infiniband-y)
+     ib_cm.o iw_cm.o ib_addr.o \
+     $(infiniband-y)
 obj-$(CONFIG_INFINIBAND_USER_MAD) += ib_umad.o
 obj-$(CONFIG_INFINIBAND_USER_ACCESS) += ib_uverbs.o ib_ucm.o \
      $(user_access-y)
-- 
1.7.1